harmony-fe/src/components/appBar/appBar.tsx

```diff
@@ -20,47 +43,97 @@ const AppBar = () => {
-            <Button href="/profile" className="navButton" color="inherit">
-              Profile
-            </Button>
-            <Button href="/home" className="navButton" color="inherit">
-              Search
-            </Button>
+  const signOut = () => {
+    cookieInfo.userCookie = getCookie('userCookie')
+    document.cookie = 'userCookie = null; expires=Thu, 18 Dec 2013 12:00:00 UTC'
+    sendHttpRequest()
+    window.location.href = '../home'
+  }
+
+  if (user === undefined) {
+    return (
+      <React.Fragment>
+        <Box sx={{ flexGrow: 1 }}>
+          <MuiAppBar className="navBar-root" position="sticky">
+            <Toolbar>
+              <Button href="/home">
+                <Box
+                  component="img"
+                  sx={{
+                    height: 64,
+                  }}
+                  alt="Harmony Logo"
+                  src={'/harmony1.png'}
+                />
+              </Button>
+              <Button href="/home" className="navButton" color="inherit">
+                Home
+              </Button>
+              <Button href="/home" className="navButton" color="inherit">
+                Search
+              </Button>
 
-            <Divider orientation="vertical" flexItem sx={{ flexGrow: 1 }} />
+              <Divider orientation="vertical" flexItem sx={{ flexGrow: 1 }} />
 
-            <Button className="navButton" color="inherit" onClick={handleOpen}>
-              New Post
-            </Button>
-            <Divider orientation="vertical" flexItem />
-            <Button href="/login" className="navButton" color="inherit">
-              Login
-            </Button>
-            <Divider orientation="vertical" flexItem />
-          </Toolbar>
-        </MuiAppBar>
-      </Box>
-      <PostModal open={open} onClose={handleClose} />
-    </React.Fragment>
-  )
+              <Divider orientation="vertical" flexItem />
+              <Button href="/login" className="navButton" color="inherit">
+                Login
+              </Button>
+              <Divider orientation="vertical" flexItem />
+            </Toolbar>
+          </MuiAppBar>
+        </Box>
+        <PostModal open={open} onClose={handleClose} />
+      </React.Fragment>
+    )
+  } else {
+    return (
+      <React.Fragment>
+        <Box sx={{ flexGrow: 1 }}>
+          <MuiAppBar className="navBar-root" position="sticky">
+            <Toolbar>
+              <Button href="/home">
+                <Box
+                  component="img"
+                  sx={{
+                    height: 64,
+                  }}
+                  alt="Harmony Logo"
+                  src={'/harmony1.png'}
+                />
+              </Button>
+              <Button href="/home" className="navButton" color="inherit">
+                Home
+              </Button>
+              <Button href="/profile" className="navButton" color="inherit">
+                Profile
+              </Button>
```

Comment:
Instead of duplicating all of this code, could we use a conditional to render the children we need if the user is logged in? 

Something like this: 
```suggestion
              {user && (<Button href="/profile" className="navButton" color="inherit">
                Profile
              </Button>)}
```

Reply:
That would totally be better, good idea.